packages/typespec-python/test/azure/generated/payload-multipart/payload/multipart/operations/_operations.py

```diff
@@ -297,16 +295,25 @@ def complex(  # pylint: disable=inconsistent-return-statements
 
         cls: ClsType[None] = kwargs.pop("cls", None)
 
-        if isinstance(body, _model_base.Model):
-            _body = handle_multipart_form_data_model(body)
-        else:
-            _body = body
-        _files = {k: multipart_file(v) for k, v in _body.items() if isinstance(v, (IOBase, bytes))}
-        _data = {k: multipart_data(v) for k, v in _body.items() if not isinstance(v, (IOBase, bytes))}
+        _body = body.as_dict() if isinstance(body, _model_base.Model) else body
+        _files: List[FilesType] = []
+        _data: Dict[str, Any] = {}
+        if _body.get("id") is not None:
+            _data["id"] = _body["id"]
+        if _body.get("address") is not None:
+            _data["address"] = json.dumps(_body["address"], cls=SdkJSONEncoder, exclude_readonly=True)
+        if _body.get("profileImage") is not None:
+            _files.append(("profileImage", _body["profileImage"]))
+        if _body.get("previousAddresses") is not None:
+            _data["previousAddresses"] = json.dumps(
+                _body["previousAddresses"], cls=SdkJSONEncoder, exclude_readonly=True
+            )
+        if _body.get("pictures") is not None:
+            _files.extend([("pictures", p) for p in _body["pictures"]])
```

Comment:
Not block review comment, I think we could make the generated code more clean if we move the logic into _vendor  like: 
```python
        file_properties = ["profileImage"]
        _data, _files = handle_multipart_form_data_body(body, file_properties)
```